nucliadb_node/nucliadb_node/pull.py

```diff
@@ -70,12 +70,19 @@ def __init__(
     async def finalize(self):
         if self.gc_task:
             self.gc_task.cancel()
+
         for subscription in self.subscriptions:
             try:
                 await subscription.drain()
             except nats.errors.ConnectionClosedError:
                 pass
         self.subscriptions = []
+
+        try:
+            await self.nc.close()
+        except (RuntimeError, AttributeError):  # pragma: no cover
```

Comment:
Can you comment why we could have `RuntimeError` and `AttributeError`

Reply:
They were handled on NATS utility. RuntimeError can occurr if the event loop is closed. I'm not sure about the AttributeError though. I'll try to know it and write down a comment